### PR TITLE
feat (examples, docs): add Nest.js example

### PR DESCRIPTION
**Pull request by lgrammel**
## Summary
- add nest.js example
- cleanup other backend framework examples
- change wording to ai sdk
- link to example sources

### CODE REVIEW

File: content/examples/15-api-servers/index.mdx [hunk 1]
```diff
@@ -1,33 +1,38 @@
...
     {
       title: 'Express',
-      description: 'Stream text to an Express server.',
+      description: 'Send AI responses from an Express server.',
       href: '/examples/api-servers/express',
     },
     {
       title: 'Hono',
-      description: 'Stream text to a Hono server.',
+      description: 'Send AI responses to a Hono server.',
```
Comment (nicoalbanese) [inline content/examples/15-api-servers/index.mdx:24]: from rather than to

File: examples/fastify/README.md [hunk 2]
```diff
@@ -1,4 +1,6 @@
-# Fastify + Vercel AI SDK Example
+# Fastify + AI SDK Example
+
+You can use the AI SDK in an [Fastify](https://fastify.dev/) server to generate and stream text and objects to the client.
```
Comment (nicoalbanese) [inline examples/fastify/README.md:3]: do we need "to the client" here?

File: examples/hono/README.md [hunk 3]
```diff
@@ -1,4 +1,6 @@
-# Hono + Vercel AI SDK Example
+# Hono + AI SDK Example
+
+You can use the AI SDK in an [Hono](https://hono.dev/) server to generate and stream text and objects to the client.
```
Comment (nicoalbanese) [inline examples/hono/README.md:3]: same as above, do we need "to the client"?

File: examples/fastify/README.md (hunk 2 above)
Comment (lgrammel) [reply examples/fastify/README.md:3]: no

**Review by nicoalbanese — APPROVED** [review]

**Pull request merged**